File: app/src/main/java/com/OdiousPanda/rainbow/API/NearbySearchCall.java
package com.OdiousPanda.rainbow.API;

import com.OdiousPanda.rainbow.DataModel.Nearby.NearbySearch;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface NearbySearchCall {
    @GET("maps/api/place/nearbysearch/json")
    Call<NearbySearch> searchNearby(@Query("location") String locationString,
                                    @Query("radius") int radius,
                                    @Query("keyword") String keyword,
                                    @Query("key") String apiKey);
}

File: app/src/main/java/com/odiousPanda/rainbowKt/apis/NearbySearchCall.kt
package com.odiousPanda.rainbowKt.apis

import com.odiousPanda.rainbowKt.model.dataSource.Nearby.NearbySearch
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

interface NearbySearchCall {
    @GET("maps/api/place/nearbysearch/json")
    suspend fun searchNearby(
        @Query("location") locationString: String?,
        @Query("radius") radius: Int,
        @Query("keyword") keyword: String?,
        @Query("key") apiKey: String?
    ): Response<NearbySearch>
}